webbpsf/roman.py

```diff
@@ -558,6 +570,89 @@ def _unmasked_pupil_path(self):
     def _masked_pupil_path(self):
         return self._pupil_controller._masked_pupil_path
 
+    def _get_filter_mode(self, wfi_filter):
+        """
+        Given a filter name, return the WFI mode
+
+        Parameters
+        ----------
+        wfi_filter : string
+            Name of WFI filter
+
+        Returns
+        -------
+        mode : string
+            Returns 'imaging', 'grism' or 'prisim' depending on filter.
+
+        Raises
+        ------
+        ValueError
+            If the input filter is not found in the WFI filter list
+        """
+
+        wfi_filter = wfi_filter.upper()
+        if wfi_filter == GRISM_FILTER:
+            return 'grism'
+        elif wfi_filter == PRISM_FILTER:
+            return 'prisim'
+        elif wfi_filter in self.filter_list:
+            return 'imaging'
+        else:
+            raise ValueError("Instrument %s doesn't have a filter called %s." % (self.name, wfi_filter))
+
+    @property
+    def mode(self):
+        """Current WFI mode"""
+        return self._get_filter_mode(self.filter)
+
+    @mode.setter
+    def mode(self, value):
+        """Mode is set by changing filters"""
+        raise AttributeError("WFI mode cannot be directly specified; WFI mode is set by changing filters.")
+
+    def override_aberrations(self, aberrations_path):
+        """Override and lock detector aberrations"""
+        self._load_detector_aberrations(aberrations_path)
+        self._aberrations_files['custom'] = aberrations_path
+        self._is_custom_aberrations = True
+
+    def reset_override_aberrations(self):
+         """Release detector aberrations override and load defaults"""
+         aberrations_path = self._aberrations_files[self.mode]
+         self._load_detector_aberrations(aberrations_path)
+         self._aberrations_files['custom'] = None
+         self._is_custom_aberrations = False
+
+    @RomanInstrument.filter.setter
+    def filter(self, value):
+
+        # Update Filter
+        # -------------
+        value = value.upper()  # force to uppercase
+
+        if value not in self.filter_list:
+            raise ValueError("Instrument %s doesn't have a filter called %s." % (self.name, value))
+
+        self._filter = value
+
+        # Update Aberrations
+        # ------------------
+        # Check if _aberrations_files has been initiated (not empty) and if aberrations are locked by user
+        if self._aberrations_files and not self._is_custom_aberrations:
```

Comment:
When the class is initialized, with `super(WFI, self).__init__("WFI", pixelscale=pixelscale)`, the filter is set to a value. We can not fill in ` self._aberrations_files` before the `super().__init__` because the `self._datapath` is not yet initialized. So if we define `self._datapath` before `super().__init__`, this will not be needed. But I thought this was the safest way forward.